dangerzone/conversion/pixels_to_pdf.py

```diff
@@ -27,137 +29,72 @@ async def convert(
         num_pages = len(glob.glob(f"{tempdir}/dangerzone/page-*.rgb"))
         total_size = 0.0
 
+        safe_doc = fitz.Document()
+
         # Convert RGB files to PDF files
         percentage_per_page = 45.0 / num_pages
-        for page in range(1, num_pages + 1):
-            filename_base = f"{tempdir}/dangerzone/page-{page}"
+        for page_num in range(1, num_pages + 1):
+            filename_base = f"{tempdir}/dangerzone/page-{page_num}"
             rgb_filename = f"{filename_base}.rgb"
             width_filename = f"{filename_base}.width"
             height_filename = f"{filename_base}.height"
-            png_filename = f"{tempdir}/page-{page}.png"
-            ocr_filename = f"{tempdir}/page-{page}"
-            pdf_filename = f"{tempdir}/page-{page}.pdf"
 
             with open(width_filename) as f:
-                width = f.read().strip()
+                width = int(f.read().strip())
             with open(height_filename) as f:
-                height = f.read().strip()
-
+                height = int(f.read().strip())
+            with open(rgb_filename, "rb") as rgb_f:
+                untrusted_rgb_data = rgb_f.read()
             # The first few operations happen on a per-page basis.
             page_size = os.path.getsize(filename_base + ".rgb") / 1024**2
             total_size += page_size
             timeout = self.calculate_timeout(page_size, 1)
 
+            self.update_progress(
+                f"Converting page {page_num}/{num_pages} from pixels to PDF"
+            )
+            pixmap = fitz.Pixmap(
+                fitz.Colorspace(fitz.CS_RGB), width, height, untrusted_rgb_data, False
```

Comment:
Let's make the `untrusted_rgb_data` explicitly as the `samples` [argument](https://pymupdf.readthedocs.io/en/latest/pixmap.html#id7) (rgb data). This is security-critical since if it interprets this data as any other format, it is not ensuring the security properties of Dangerzone.

Reply:
We should also add a test to ensure that this does not accept any other format. For example, we could pass it a png and ensure that fails.

Reply:
Well, I guess we should pass those as non-keyword arguments. They way they are defined is a bit quirky after all:

https://github.com/pymupdf/PyMuPDF/blob/5a000bc8bda0293de74b00b33b3610f94cc68b94/fitz/fitz.i#L7761